### PR TITLE
Randomly choose workflow to run in GH action.

### PR DESCRIPTION
**Pull request by revit13**
## Why are these changes needed?
To cover more workflow runs this PR chooses the workflow to run randomly.
In addition a new GH action is created to manually run workflows.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -112,4 +112,7 @@ jobs:
                   make -C kind setup
                   make -C kfp/kfp_support_lib test
                   make -C transforms workflow-build
-                  make -C transforms/universal/noop workflow-test
+                  dir=("code"  "universal") && index=$(($RANDOM % 2)) && subdirs=${dir[$index]} && transforms=($(find transforms/$subdirs/ -type d  -maxdepth 1 ))
+                  # First element is not really a subdir but rather the current dir so remove it and randomly choose a transform to run
+                  set -- "${transforms[@]}" && shift && transforms=("$@") && size=${#transforms[@]} && index=$(($RANDOM % $size))
+                  make -C ${transforms[$index]} workflow-test
```
Comment (roytman) [inline .github/workflows/test.yml]: does it print the chosen workflow name?

File: .github/workflows/workflow_manual_run.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+              timeout-minutes: 120
+              run: |
+                  source kind/requirements.env
+                  export PATH=$PATH:/tmp/
+                  curl -Lo /tmp/kind https://kind.sigs.k8s.io/dl/v${KIND_VERSION}/kind-linux-amd64
+                  curl -fsSL -o /tmp/get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3
+                  chmod 700 /tmp/get_helm.sh
+                  HELM_INSTALL_DIR=/tmp/ /tmp/get_helm.sh -v v${HELM_VERSION} --no-sudo
+                  chmod 777 /tmp/helm
+                  chmod 777 /tmp/kind
```
Comment (roytman) [inline .github/workflows/workflow_manual_run.yml]: can you reorder command, so the chmod will be near the dir/file creation?

File: .github/workflows/test.yml (hunk 1 above)
Comment (revit13) [reply .github/workflows/test.yml]: Yes. please see the following examples: https://github.com/revit13/data-prep-lab/actions/workflows/debug_workflow.yml

Comment (roytman) [reply .github/workflows/test.yml]: I don't see it at https://github.com/revit13/data-prep-lab/actions/runs/9535702057/job/26281751327, after all `kubectl describe pods` ...

Comment (revit13) [reply .github/workflows/test.yml]: Sorry i put the link to the manual run by mistake... please see the end of: `Test CI / test-kfp-workflow-run (pull_request) Successful in 20m` run in this PR

```
12:45:14 INFO - Pipeline uploaded
12:45:14 INFO - Pipeline run filter_wf 2024_06_16_12_45_14 submitted
12:46:14 INFO - Got pipeline execution status Running
12:47:14 INFO - Got pipeline execution status Succeeded
12:47:14 INFO - Pipeline filter_wf successfully completed
```

File: .github/workflows/workflow_manual_run.yml (hunk 2 above)
Comment (revit13) [reply .github/workflows/workflow_manual_run.yml]: Fixed. Thanks

**Review by roytman — APPROVED** [review]
LGTM

**Pull request merged**